src/languages/csharp.ts

```diff
@@ -218,6 +220,10 @@ const nodeMatchers: Partial<Record<ScopeType, NodeMatcherAlternative>> = {
   ifStatement: "if_statement",
   class: "class_declaration",
   className: "class_declaration[name]",
+  condition: cascadingMatcher(
+    conditionMatcher("*[condition]"),
+    patternMatcher("~binary_expression.binary_expression!")
```

Comment:
This seems odd. Wouldn't this match eg the right hand side of `a = b + c`?

Reply:
Ah, interesting. Could you add equivalent tests to the ones I requested above for while loops too?